python_ta/reporters/color_reporter.py

```diff
@@ -57,6 +57,12 @@ def print_messages(self, level='all'):
 
         print(result)
 
+        # reset messages
+        self.reset_messages()
```

Comment:
Make the call to `reset_messages` in the client code, not in the body of `print_messages`. You should override `reset_messages` in this class to include the following two lines.